File: Irrgarten/src/irrgarten/TestP3.java

package irrgarten;

/**
 * @brief Test para la práctica 3
 * @author luisvaldivieso
 */
public class TestP3 {
    private static java.util.Random generator = new java.util.Random();
    
    public static void main(String args[]){
        Game game = new Game(3);
        Directions [] directions = new Directions[4];
        directions[0]=Directions.DOWN;
        directions[1]=Directions.UP;
        directions[2]=Directions.LEFT;
        directions[3]=Directions.RIGHT;
        
        GameState gameState;
        boolean endGame = false;
        int rounds = 0;
        while (!endGame && rounds < 10){
            Directions generatedDirection = directions[generator.nextInt(4)];
            
            game.nextStep(generatedDirection);
            
            gameState = game.getGameState();
            System.out.println(gameState.getLabyrinth().toString());
            System.out.println("Current Player: " + gameState.getCurrentPlayer());
            System.out.println("Chosen direction: " + generatedDirection.toString());
            System.out.println(gameState.getLog());
            System.out.println(gameState.getMonsters());
            System.out.println(gameState.getPlayers());
            rounds++;
        }
        
        
        
    }
}

File: Irrgarten_ruby/test_p3.rb
# encoding:utf-8
require_relative 'directions'
require_relative 'game'

class TestP3
    def self.main
        game = Irrgarten::Game.new(3)
        directions = [Irrgarten::Directions::DOWN, Irrgarten::Directions::UP, Irrgarten::Directions::LEFT, Irrgarten::Directions::RIGHT]

        end_game = false
        rounds = 0
        while !end_game && rounds < 10
            generated_direction = directions[rand(4)]


            game_state = game.game_state
            puts game_state.labyrinth.to_s
            puts "Current Player: #{game_state.current_player}"
            puts "Chosen direction: #{generated_direction}"
            puts game_state.log
            puts game_state.monsters
            puts game_state.players
            game.next_step(generated_direction)
            rounds += 1
        end
    end
end

TestP3.main